snorkel/labeling/preprocess/core.py

```diff
@@ -58,9 +56,11 @@ def preprocess(self, **kwargs: Any) -> FieldMap:
     def __call__(self, x: DataPoint) -> DataPoint:
         field_map = {k: getattr(x, v) for k, v in self.field_names.items()}
         preprocessed_fields = self.preprocess(**field_map)
-        preprocessed_fields = {
-            v: preprocessed_fields[k] for k, v in self.preprocessed_field_names.items()
-        }
+        if self.preprocessed_field_names is not None:
+            preprocessed_fields = {
+                v: preprocessed_fields[k]
+                for k, v in self.preprocessed_field_names.items()
+            }
```

Comment:
Some of my comments are redundant with the other PR- I think there should be a check, since this depends on user returning a dict-type?